### PR TITLE
Raise explicit exception for geometry collections

### PR DESCRIPTION
**Pull request by rmarianski**
Connects to https://github.com/mapzen/mapbox-vector-tile/issues/50

@zerebubuth could you review please?

### CODE REVIEW

**Comment by zerebubuth** [comment]
:+1:

Would it also be worth mentioning in the README that we don't support geometry collections?

File: mapbox_vector_tile/encoder.py [hunk 1]
```diff
@@ -229,6 +229,11 @@ def _get_feature_type(self, shape):
             return self.tile.LineString
         elif shape.type == 'Polygon' or shape.type == 'MultiPolygon':
             return self.tile.Polygon
+        elif shape.type == 'GeometryCollection':
+            raise ValueError('Encoding geometry collections not supported')
+        else:
+            raise ValueError('Cannot encode unknown geometry type: %s' %
+                             shape.type)
```
Comment (loicgasser) [inline mapbox_vector_tile/encoder.py]: I think this is dead code as shapely only supports:
Point, MultiPoint, LineString, MultiLineString, Polygon, MultiPolygon

I guess that's why you can't create a test for it also.

Comment (rmarianski) [inline mapbox_vector_tile/encoder.py]: I think so too, but I'd rather have dead code that would explicitly fail loudly this way, if only to make the code clearer. I thought about using an assertion instead, but opted for a `ValueError`. Do you think an assertion would have been more appropriate?

**Comment by rmarianski** [comment]
Added a note in the readme in 43d589d.

**Pull request merged**